crates/adapters/src/format/avro/output.rs

```diff
@@ -75,6 +75,16 @@ impl OutputFormat for AvroOutputFormat {
             )
         })?;
 
+        if matches!(
```

Comment:
I apologize about the confusion. I didn't mean to suggest to do this in every format encoder. Probably the right way to solve this would be to pass the entire endpoint configuration to each transport connector, and have that validate the format used. This can be a follow-up PR.